app/controllers/books_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all
+  end
+
+  def create
+    @book = Book.new(post_params)
+
+    if @book.save
+      # render success in Jbuilder
+    else
+      render json: { message: "400 Bad Request" }, status: :bad_request
+    end
+  end
+
+  def show
+    @book = Book.find_by_id(params[:id])
+
+    if @book.nil?
```

Comment:
Sorry still don't get it.. 
```
if @book.nil?
      render json: { message: "Cannot find book" }, status: :not_found
    end
```
Are we talking about this?
Both index and create functions dont require finding a book. Can I still use before_action?

```
before_action :authenticate_user!, :except => [:index]
```
I have added this in my next pull request